lib/spack/docs/Makefile

```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -E
+SPHINXOPTS    = -W
```

Comment:
What's up with this change? Don't we want the documentation tests to fail if there are any warnings?

Reply:
We do want this.  This is how RTD actually runs the docs so if there is an error, we'll catch it before the docs break now.

Reply:
But doesn't `-E` cause an error and `-W` only cause a warning?

Reply:
No, from `sphinx-build -h`:

```
  -E                don't use a saved environment, always read all files
  -W                turn warnings into errors
```